build/Dockerfile.builder

```diff
@@ -3,19 +3,30 @@ FROM debian:buster as bazel-builder
 COPY build/install-bazel.sh /build/
 RUN chmod +x /build/install-bazel.sh && /build/install-bazel.sh
 
-# Build auth binary.
+# Copy in only the necessary files for building.
 FROM bazel-builder as auth-builder
-COPY . /src
+COPY Makefile WORKSPACE /src/
+COPY bazel /src/bazel
+COPY config /src/config
+COPY src /src/src
+COPY test /src/test
+
+# Build auth binary.
 WORKDIR /src
 RUN bazel build //...
 
 # Create our final auth-server container image.
 FROM debian:buster
 RUN groupadd -r auth-server-grp && useradd -m -g auth-server-grp auth-server-usr
 
-COPY --from=auth-builder /src/bazel-bin/src/main/auth-server /app/auth-server
-RUN chgrp auth-server-grp /app/auth-server && chown auth-server-usr /app/auth-server && chmod u+x /app/auth-server
+COPY --from=auth-builder /src/bazel-bin/src/main/auth_server /app/auth_server
```

Comment:
Can we collapse these `COPY` statements into 1 to reduce the number of layers.

Reply:
I originally put the boost libraries into a single copy with a wildcard and it didn't seem to like it for some reason. I thought maybe it didn't like wildcard in `COPY --from` statements but it looks like I've used a wildcard in one in another project so I'll have another go and see if I can figure out why it's not working. Or at least I could just put all the separate filenames into a single copy.

Reply:
Couldn't get it to work with a wildcard for some reason, it just omitted copying the files, but I've combined them all into a single COPY statement just listing each file separately.